_posts/2018-07-20-snowplow-mini-0.6.0-released/index.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title: Snowplow Mini 0.6.0 released
+title-short: Snowplow Mini 0.6.0
+tags: [snowplow-mini, docker, iglu-server, elasticsearch]
+author: Oguzhan
+category: Releases
+permalink: /blog/2018/07/20/snowplow-mini-0.6.0-released/
+---
+
+We are pleased to announce the 0.6.0 release of Snowplow Mini, our accessible "Snowplow in a box" distribution.
+
+Snowplow Mini is the complete Snowplow real-time pipeline running on a single instance, available
+for easy deployment as a pre-built AMI. Use it to:
+
+1. Set up an inexpensive and easily discardable Snowplow stack for testing your tracker and schema changes
+2. Learn about Snowplow without having to set up a horizontally-scalable, highly-available production-grade pipeline
+
+This release focuses on addressing the long-standing [migration to Docker][docker-migration-issue] including updating Elasticsearch stack. Also, as of this release, Snowplow Mini will be available in 3 different sizes; `large`, `xlarge` and `xxlarge` to meet varying purposes.
+
+Read on for:
+
+<!--more-->
+
+1. [New Iglu Server](#iglu-server)
+2. [Docker migration](#docker-migration)
+3. [Google Cloud Platform support](#gcp-support)
+4. [Other updates](#other-changes)
+5. [Documentation and getting help](#help)
+
+
+<h2 id="iglu-server">1. New Iglu Server</h2>
+
+One of our goals for Snowplow Mini is making it stateless, meaning that all the required services such as Iglu Server, Elasticsearch, Postgres etc live outside the actual box running Snowplow Mini.
+
+As part of this goal, we've introduced plenty of features for Control Plane previously in [Snowplow Mini 0.4.0][snowplow-mini-0.4.0-post]. Today we are adding a new feature on top of them, enabling Iglu Server to use an external Postgres instance.
+
+Instead of specifying external Postgres configuration only, placed in Iglu Server's configuration file, we introduce the ability to upload Iglu Server configuration file, enabling to play with all bits of the configuration including Postgres connection details.
+
+![iglu-server-conf][iglu-server-conf-img]
+
+Note that this release also bumps Iglu Server to `0.3.0` which [introduced][iglu-server-improvements] a new configuration parameter `repo-server.baseURL` meaning that our users should upload their own Iglu Server config file with `repo-server.baseURL` set to `<snowplow-mini-deployment-address>/iglu-server`, if they want to use Swagger UI of Iglu Server.
```

Comment:
I'm not sure I understand this statement

> if they want to use Swagger UI of Iglu Server.

Does it mean users won't have Swagger UI without uploading own config file?

Reply:
Just as you understand @chuwy , Swagger UI won't work unless you set `baseURL` to your deployment address.

Reply:
Could it be (theoretically) automated?

Reply:
@chuwy I believe so, a simple button on Control Plane to send `window.location.host` to backend could set baseURL for Sp Mini users.

Reply:
It is not as automated as you want I guess.

Reply:
couldn't we set that up when the user adds the master api key?

Reply:
Sure we can @BenFradet . We just need to inject this into somewhere. Adding master api key is mandatory and a single operation most of the time. Great place for injecting this baseURL set operation.

Reply:
I agree that it is very important to reduce amount of manual steps. On the other hand, Swagger UI is not the most popular feature of Snowplow Mini (assuming rest of Iglu Server works as is, which is absolutely crucial)

Reply:
Yes @chuwy , you can use Iglu Server as is by pointing your curl requests at `<deployment-address>/iglu-server` . You don't need to set `baseURL` for usages other than Swagger UI.

Reply:
How much effort would it be to automate? I agree we should if we can...

Reply:
Not to add any additional setup step (including clicking on a new button etc), we can inject baseURL setup into one of already existing steps. Considering it needs to be done once, I think it is best to inject into `add api key for local Iglu Server`, which needs to be done once as well.